FluentTerminal.Models/AbstractCommand.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+using System.ComponentModel;
+using FluentTerminal.Models.Enums;
+
+namespace FluentTerminal.Models
+{
+    /// <summary>
+    /// Wrapper class to abstract handling both static enum based commands, as well as dynamic commands such as per-shell shortcuts.
+    /// </summary>
+    public abstract class AbstractCommand
+    {
+        public abstract string Description { get; }
+        public abstract override int GetHashCode();
+        public abstract override bool Equals(object obj);
+        public abstract override string ToString();
+
+        public static bool operator ==(AbstractCommand A, AbstractCommand B)
+        {
+            return A.Equals(B);
+        }
+
+        public static bool operator !=(AbstractCommand A, AbstractCommand B)
+        {
+            return !A.Equals(B);
+        }
+
+        public static implicit operator AbstractCommand(Command input)
+        {
+            return new EnumCommand<Command>(input);
```

Comment:
It looks really interesting, but I guess its breaking the abstracting if the abstract class nows about its derived classes

Reply:
I can have a look at it this weekend